java/adapter/avro/src/test/java/org/apache/arrow/AvroTestBase.java

```diff
@@ -45,19 +46,17 @@
 import org.apache.avro.io.DatumWriter;
 import org.apache.avro.io.DecoderFactory;
 import org.apache.avro.io.EncoderFactory;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.BeforeEach;
 
 public class AvroTestBase {
 
-  @ClassRule
-  public static final TemporaryFolder TMP = new TemporaryFolder();
+  public Path tmp;
 
   protected AvroToArrowConfig config;
 
-  @Before
-  public void init() {
+  @BeforeEach
+  void prepare() throws Exception {
+    this.tmp = Files.createTempDirectory("avro");
```

Comment:
Hmm, this doesn't remove the temporary directory though, and we don't want to leave a bunch of clutter on the filesystem

Reply:
@lidavidm Removing such temporary directory is what causing an error on Windows, the directory is still being used by some thread at the moment of  `@AfterEach`.  I suspect there could be a leaked file handler somewhere in the code, but I don't have a Windows machine to debug this.  Polluting the file system should not be a problem, since OS cleans up temporary storage periodically. 
Or we could keep `@TemporaryFolder` from Junit 4 in these tests until  https://github.com/junit-team/junit5/issues/2811 is resolved

Reply:
tmpfs isn't cleaned up on Linux, at least, not until reboot. It might be easier to keep the JUnit4 implementation then.

Reply:
@lidavidm Found another work around with `deleteOnExit`